synapse/cortex.py

```diff
@@ -1625,13 +1627,93 @@ def addLayerCtor(self, name, ctor):
 
     async def _initCoreViews(self):
 
+        defiden = self.cellinfo.get('defaultview')
+
         for iden, node in await self.hive.open(('cortex', 'views')):
             view = await s_view.View.anit(self, node)
             self.views[iden] = view
+            if iden == defiden:
+                self.view = view
+
+        # if we have no views, we are initializing.  Add a default main view and layer.
+        if not self.views:
+            layr = await self.addLayer()
+            iden = s_common.guid()
+            view = await self.addView(iden, 'root', (layr.iden,))
+            await self.cellinfo.set('defaultview', iden)
+            self.view = view
+
+    async def _moveDict(self, oldpath, newpath):
+        '''
+        Move a dict-only hive node from old path to new path, doing nothing if old path doesn't exist
+
+        Returns:
+            True if the old path was present
+        '''
+        if not await self.hive.exists(oldpath):
+            return False
+
+        oldnode = await self.hive.open(oldpath)
+        oldinfo = await oldnode.dict()
+
+        newnode = await self.hive.open(newpath)
+        newinfo = await newnode.dict()
+
+        for name, valu in oldinfo.items():
+            await newinfo.set(name, valu)
+
+        await oldnode.pop(())
+        return True
+
+    async def _migrateViewsLayers(self):
+        '''
```

Comment:
We need to add a regression test for this using one the synapse-regression cortexes.

Reply:
done